graphql.go

```diff
@@ -33,18 +33,36 @@ func NewClient(url string, httpClient *http.Client) *Client {
 // with a query derived from q, populating the response into it.
 // q should be a pointer to struct that corresponds to the GraphQL schema.
 func (c *Client) Query(ctx context.Context, q any, variables map[string]any) error {
-	return c.do(ctx, queryOperation, q, variables)
+	return c.do(ctx, queryOperation, q, variables, nil)
+}
+
+// QueryWithExtensions executes a single GraphQL query request,
+// with a query derived from q, populating the response into it.
+// q should be a pointer to struct that corresponds to the GraphQL schema.
+// Additionally, this will capture the extensions from the response.
+// extensions should be a pointer that corresponds to the extensions schema.
+func (c *Client) QueryWithExtensions(ctx context.Context, q interface{}, variables map[string]interface{}, extensions interface{}) error {
```

Comment:
nit: here you use `interface{}` but in the client you use `any`

Reply:
Will implement